tests/unit/conftest.py

```diff
@@ -8,10 +8,13 @@
 
 
 @pytest.fixture(autouse=True)
-def patched_idle(mocker):
-    yield mocker.patch(
-        "events.upgrade.ZKUpgradeEvents.idle", new_callable=PropertyMock, return_value=True
-    )
+def patched_idle(mocker, request):
+    if "nopatched_idle" in request.keywords:
+        yield
+    else:
+        yield mocker.patch(
+            "events.upgrade.ZKUpgradeEvents.idle", new_callable=PropertyMock, return_value=True
+        )
```

Comment:
apology: yeah this sucks, I looked at it for a while several PRs back and hate every option out there. Really surprised it isn't easier. 

The other option is to locally patch the fixture with a new one of the same name, which might be neater. Up to you though!

Reply:
I did not want to patch `idle` (in the tests I'm just setting the upgrade stack). That's actually how I managed to discover that mismatch of handling the `idle` property among charms. If I had mocked `idle` directly, I would not have found that out